docs/guide/setup/defining_protocols.rst

```diff
@@ -0,0 +1,65 @@
+Defining Protocols
```

Comment:
I'm not sure this title really matches the contents of what is being presented here.

Possibly could be called "Creating and using Protocols"?